modules/data-streams/src/main/java/org/elasticsearch/datastreams/lifecycle/DataStreamLifecycleService.java

```diff
@@ -310,7 +330,168 @@ void run(ClusterState state) {
+                    if (downsampleSourceIndex != null) {
+                        affectedIndices.add(downsampleSourceIndex.getIndex());
+                        // delete downsampling source index (that's not part of the data stream anymore) before doing any more
+                        // downsampling
+                        DeleteIndexRequest deleteIndexRequest = new DeleteIndexRequest(backingIndexDownsamplingSource).masterNodeTimeout(
+                            TimeValue.MAX_VALUE
+                        );
+                        transportActionsDeduplicator.executeOnce(
+                            deleteIndexRequest,
+                            new ErrorRecordingActionListener(backingIndexDownsamplingSource, errorStore),
+                            (req, reqListener) -> deleteIndex(
+                                deleteIndexRequest,
+                                "replacement with its downsampled index in the data stream",
+                                reqListener
+                            )
+                        );
+                        continue;
+                    }
+                }
+
+                // this index has matching downsample rounds and is read-only, so let's wait for an in-progress downsampling operation to
+                // succeed or trigger the corresponding one
+                // TODO add support for cancelling a current in-progress operation if another, later, round matches
+                DataStreamLifecycle.Downsampling.Round lastRound = downsamplingRounds.get(downsamplingRounds.size() - 1);
+
+                for (DataStreamLifecycle.Downsampling.Round round : downsamplingRounds) {
+                    // the downsample index name for each round is deterministic
+                    String downsampleIndexName = DownsampleConfig.generateDownsampleIndexName(
+                        DOWNSAMPLED_INDEX_PREFIX,
+                        backingIndex,
+                        round.config().getFixedInterval()
+                    );
+                    IndexMetadata downsampleIndexMeta = metadata.index(downsampleIndexName);
+                    if (downsampleIndexMeta != null) {
+                        IndexMetadata.DownsampleTaskStatus downsampleStatus = INDEX_DOWNSAMPLE_STATUS.get(
+                            downsampleIndexMeta.getSettings()
+                        );
+                        if (downsampleStatus.equals(UNKNOWN) && round.equals(lastRound)) {
+                            // target downsampling index exists and is not a downsampling index (name clash?)
+                            // we fail now but perhaps we should just randomise the name?
+                            errorStore.recordError(
+                                backingIndex.getIndex().getName(),
+                                new ResourceAlreadyExistsException(downsampleIndexMeta.getIndex().getName())
+                            );
```

Comment:
Why doesn't this one have a `break;` when the others do? If we are in an unknown state we should continue with the other rounds of downsampling?

Reply:
It's because we only record this error for the last round of downsampling (we'll ever only look to _start_ a new downsampling round for the last matching round)